### PR TITLE
docs: Clarify coordination for backporting process

### PR DESCRIPTION
**Pull request by christarazi**
Document the common workflow that we've been working under.

Signed-off-by: Chris Tarazi <chris@isovalent.com>

### CODE REVIEW

File: Documentation/contributing/development/contributing_guide.rst [hunk 1]
```diff
@@ -473,6 +473,27 @@ Follow the :ref:`backport_process` guide to know how to perform this task.
...
 
+Coordination
+++++++++++++
+
+In general, coordinating in the #launchpad channel with your colleague is
+encouraged. It can reduce your workload and it will avoid backporting conflicts
+such as opening a PR with the same backports. Such discussions will typically
+revolve around which branches to tackle and which day of the week.
+
+An example interaction is #launchpad:
```
Comment (qmonnet) [inline Documentation/contributing/development/contributing_guide.rst]:
```suggestion
An example interaction in #launchpad:
```

File: Documentation/contributing/development/contributing_guide.rst [hunk 2]
```diff
@@ -473,6 +473,27 @@ Follow the :ref:`backport_process` guide to know how to perform this task.
...
+
+::
+
+    Starting backport round for v1.7 and v1.8 now
+    /cc @your-actual-colleague
```
Comment (qmonnet) [inline Documentation/contributing/development/contributing_guide.rst]: Nit: Won't Slack complain that `/cc` is an unknown command? :)

File: Documentation/contributing/development/contributing_guide.rst [hunk 3]
```diff
@@ -473,6 +473,27 @@ Follow the :ref:`backport_process` guide to know how to perform this task.
 
 .. _dev_coo:
 
+Coordination
+++++++++++++
+
+In general, coordinating in the #launchpad channel with your colleague is
```
Comment (qmonnet) [inline Documentation/contributing/development/contributing_guide.rst]:
```suggestion
In general, coordinating in the #launchpad Slack channel with your colleague is
```

**Review by qmonnet — APPROVED** [review]
Non-blocking nits

**Review by joestringer — COMMENTED** [review]
nit, who is "colleague" in this? How does someone figure that out? Do we mean "other hat owner for the week"?

**Comment by christarazi** [comment]
@joestringer Good point, updated

**Review by joestringer — APPROVED** [review]
Looks good. I'd like to take it even maybe one step further, which is to say "Please coordinate with the previous / next hat to figure out how to get the backport PRs merged" so we don't get in this situation where there are 3 different PRs from 3 different weeks and none of them make progress because each hat owner works in a silo and gets stuck.

**Comment by christarazi** [comment]
Another good point; I added a small paragraph describing as such.

File: Documentation/contributing/development/contributing_guide.rst [hunk 4]
```diff
@@ -473,6 +473,33 @@ Follow the :ref:`backport_process` guide to know how to perform this task.
...
+The other hat owner will then end up doing v1.9 and v1.10 the next day, for
+example.
+
+If there are many backports to be done, then splitting up the rounds can be
+beneficial. Typically, backporters opt to start a round in the beginning of the
+week and then another near the end of the week.
+
+By the start / end of the week, if there are other backport PRs that haven't
+been merged, then please coordinate with the previous / next backporter to get
+make progress on them.
```
Comment (joestringer) [inline Documentation/contributing/development/contributing_guide.rst]:
```suggestion
been merged, then please coordinate with the previous / next backporter to check
what the status is and establish who will work on getting the backports into the tree
(for instance by investigating CI failures and addressing review feedback).
```

File: Documentation/contributing/development/contributing_guide.rst [hunk 5]
```diff
@@ -473,6 +473,33 @@ Follow the :ref:`backport_process` guide to know how to perform this task.
...
+
+An example interaction in #launchpad:
+
+::
+
+    Starting backport round for v1.7 and v1.8 now
+    cc @other-hat-wearer
+
+The other hat owner will then end up doing v1.9 and v1.10 the next day, for
+example.
```
Comment (joestringer) [inline Documentation/contributing/development/contributing_guide.rst]:
```suggestion
The other hat owner can then handle v1.9 and v1.10 backports the next day, for
example.
```

**Review by joestringer — CHANGES_REQUESTED** [review]
Getting there, just a few more minor wording nits.

File: Documentation/contributing/development/contributing_guide.rst (hunk 4 above)
Comment (joestringer) [reply Documentation/contributing/development/contributing_guide.rst]: ^ I realize it's a bit ambiguous here on "who" but from a high level the backports need to get done and release folks/maintainers don't care who does it as long as it gets done. I think it's useful to have some leeway for backporters of different weeks to negotiate who will work on the backports depending on who has the available time to do so.

Comment (christarazi) [reply Documentation/contributing/development/contributing_guide.rst]: I've added a blurb mentioning the leeway and negotiation  as well

**Review by joestringer — APPROVED** [review]

**Review by errordeveloper — APPROVED** [review]
Thank you!

**Comment by nbusseneau** [comment]
@christarazi I don't think we are maintaining 1.7 anymore, are we? Shall we remove the `needs-backport/1.7` label?

**Comment by nbusseneau** [comment]
Confirmed here on public Slack, 1.7 is EOL: https://cilium.slack.com/archives/CDD1S4BHC/p1621620861010800?thread_ts=1621620829.010600&cid=CDD1S4BHC

**Pull request merged**